client/state/rewind/browser/reducer.ts

```diff
@@ -0,0 +1,198 @@
+	if ( fullPath.length > 0 && fullPath[ 0 ] === '/' ) {
+		fullPath.shift();
+	}
+
+	for ( const pathPart of fullPath ) {
+		const childNode = currentNode.children.find(
+			( node: BackupBrowserItem ) => node.path === pathPart
+		);
+		if ( ! childNode ) {
+			return result;
+		}
+		result.parent = currentNode;
+		result.index = currentNode.children.indexOf( childNode );
+
+		currentNode = childNode;
+	}
+	return result;
+};
+
+const updateChildrenStatus = (
+	nodeToUpdate: BackupBrowserItem,
+	status: 'checked' | 'unchecked' | 'mixed'
+) => {
+	for ( let i = 0; i < nodeToUpdate.children.length; i++ ) {
+		const newChild = { ...nodeToUpdate.children[ i ] };
+		newChild.checkState = status;
+		nodeToUpdate.children[ i ] = newChild;
+		if ( newChild.childrenLoaded && newChild.children.length > 0 ) {
+			updateChildrenStatus( newChild, status );
+		}
+	}
+};
+
+// Update the status of the current node based on any children
+const getCheckedStatus = ( nodeToIterate: BackupBrowserItem ) => {
+	let isMixed = false;
+	let isChecked = false;
+	let isUnchecked = false;
+	nodeToIterate.children.forEach( ( child: BackupBrowserItem ) => {
+		if ( child.checkState === 'mixed' ) {
+			isMixed = true;
+		} else if ( child.checkState === 'checked' ) {
+			isChecked = true;
+		} else {
+			isUnchecked = true;
+		}
+	} );
+	if ( isMixed ) {
+		return 'mixed';
+	} else if ( isChecked && isUnchecked ) {
+		return 'mixed';
+	} else if ( isChecked ) {
+		return 'checked';
+	}
+	return 'unchecked';
+};
+
+const updateParent = ( state: AppState, node: BackupBrowserItem ): AppState => {
+	if ( node.path === '/' ) {
+		return state;
+	}
+	const nodePath = [ ...node.ancestors ];
+	const { parent: parentOfParent, index } = getParentAndIndex( state, nodePath );
+	// If we're at the root node we'll just return
+	if ( parentOfParent === undefined || index === undefined ) {
+		return state;
+	}
+	const newNode = { ...parentOfParent.children[ index ] };
+	newNode.checkState = getCheckedStatus( newNode );
+	parentOfParent.children[ index ] = newNode;
+	return updateParent( state, newNode );
+};
+
+export default ( state = initialState, { type, payload }: AnyAction ) => {
+	switch ( type ) {
+		case JETPACK_BACKUP_BROWSER_SET_CHECK_STATE: {
+			const newState = { ...state };
+			// Payload needs to give us the path to the node and the new checkState
+			const { nodePath, checkState } = payload;
+			// We need to find the node in the tree
+			// parent.children[0]
+			const { parent, index } = getParentAndIndex( newState, nodePath );
+			if ( ! parent || index === undefined ) {
+				return state;
+			}
+			const newNode = { ...parent.children[ index ] };
+			const nodeToUpdate = getNode( state, nodePath );
+			if ( ! nodeToUpdate ) {
+				return state;
+			}
+			// We need to update the node's checkState
+			newNode.checkState = checkState;
+			parent.children[ index ] = newNode;
+			// parent.children[0] = { ...nodeToUpdate };
```

Comment:
some old/test code?

Reply:
Yeah, the comment was just some notes.  Removing